bgpd/bgp_evpn.c

```diff
@@ -1504,7 +1504,7 @@ static int update_evpn_type5_route(struct bgp *bgp_vrf, struct prefix_evpn *evp,
 	struct bgp *bgp_def = NULL;
 	int route_changed = 0;
 
-	bgp_def = bgp_get_default();
+	bgp_def = bgp_get_evpn();
```

Comment:
@vivek-cumulus  raised an excellent review point here that today we don't need to configure "advertise-all-vni" on border routers but this change will break that setup. We need a fix for that before pulling in this change.

Reply:
That is true, thanks to him for pointing this out.

The main issue is that we still need the EVPN VRF for this, and no means of determining it other than `advertise_all_vni` currently exists which globally enables EVPN.

Since the real meaning of the EVPN VRF is "the VRF with the BGP-EVPN session", should we add a flag in vrf like `evpn_enabled` which is true when either the VRF has a neighbor enabled in the `l2vpn evpn` AF, or when it has `advertise_all_vni` (for compatibility with existing configurations)? Would like your opinion on this!